packages/shared/src/types/utils.ts

```diff
@@ -0,0 +1,20 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// Useful when we want to create type aliases for types that are just a string
+export type StringAlias = { _?: never }
```

Comment:
I don't get it. Could you explain on Slack / on a call? 🙃

Reply:
Sure!

Reply:
We have a type somewhere:

```ts
type AddressHash = string
```

Now, assume there’s a prop:

```
addressHash: AddressHash
```

When you hover your mouse over it you will see screenshot 1.

But if we force TypeScript to think that this is not simply a string, then we can trick it to display the actual alias that we want (screenshot 2)

![image](https://github.com/user-attachments/assets/3ad1289a-320f-4c3f-a857-e4b3000793d0)

![image](https://github.com/user-attachments/assets/5458efdb-bb2b-4369-b6f6-b8f70388feb2)